schematic/utils/general.py

```diff
@@ -57,24 +60,66 @@ def get_dir_size(path: str):
                 total += get_dir_size(entry.path)
     return total
 
+def calculate_datetime(minutes: int, input_date: datetime, before_or_after: str = "before") -> datetime:
+    """calculate date time 
+
+    Args:
+        input_date (datetime): date time object provided by users
+        minutes (int): number of minutes
+        before_or_after (str): default to "before". if "before", calculate x minutes before current date time. if "after", calculate x minutes after current date time. 
+
+    Returns:
+        datetime:  return result of date time calculation
+    """
+    if before_or_after=="before": 
+        date_time_result = input_date - timedelta(minutes=minutes)
+    else:
+        date_time_result = input_date + timedelta(minutes=minutes)
+    return date_time_result
+
+
+def check_synapse_cache_size(directory='/root/.synapseCache')-> Union[float, int]:
```

Comment:
I found the previous way of calculating size of .synapseCache a bit confusing. When I debugged the space limit error, I used `du -sh <path to .synapseCache>` command, so I think I should make it consistent and use this command when I calculate cache